client/src/app/core/core-services/operator.service.ts

```diff
@@ -374,22 +375,27 @@ export class OperatorService {
      * Saves the current WhoAmI to storage with the updated permissions
      */
     private calcPermissions(): Permission[] {
-        let permissions;
-        if (!this.groupIds) {
-            permissions = [];
-        } else if (this.isAnonymous || this.groupIds.length === 0) {
-            // Anonymous or users in the default group.
-            permissions = this.activeMeeting?.default_group?.permissions || [];
+        const permissionSet = new Set<Permission>();
+        if (this.isAnonymous) {
+            // Anonymous is always in the default group.
+            this.activeMeeting?.default_group?.permissions.forEach(perm => permissionSet.add(perm));
         } else {
-            const permissionSet = new Set<string>();
-            this.DS.getMany(Group, this.groupIds).forEach(group => {
-                group.permissions?.forEach(permission => {
-                    permissionSet.add(permission);
+            if (this.groupIds) {
```

Comment:
`group_ids(meetingId?: Id): Id[]` (ViewUser)
will in doubt return some array, where `!![]` is true.

Replace with: `this.groupIds?.length`

Reply:
Sure, although even for `[]` the code should still work since `forEach` will just not execute ;)